src/main/java/frc/robot/subsystems/USBCamera.java

```diff
@@ -0,0 +1,86 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.CameraServer;
+import edu.wpi.cscore.UsbCamera;
+import edu.wpi.cscore.VideoSource;
+import edu.wpi.cscore.VideoSink;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+import frc.robot.Robot;
+
+/**
+ * Add your docs here.
+ */
+public class USBCamera extends Subsystem {
+  // Put methods for controlling this subsystem
+  // here. Call these from Commands.
+
+  private UsbCamera m_camera1;
+  private UsbCamera m_camera2;
+  private VideoSink server;
+  public volatile int whichCamera;
+  
+
+
+  public USBCamera(){
+    m_camera1 = CameraServer.getInstance().startAutomaticCapture();
+    m_camera1.setResolution(160, 120);
+    m_camera1.setFPS(24);
+    m_camera1.setConnectionStrategy(VideoSource.ConnectionStrategy.kKeepOpen);
+
+    m_camera2 = CameraServer.getInstance().startAutomaticCapture(1);
+    m_camera2.setResolution(160, 120);
+    m_camera2.setFPS(24);
+    m_camera2.setConnectionStrategy(VideoSource.ConnectionStrategy.kKeepOpen);
+
+    server = CameraServer.getInstance().getServer();
+
+    whichCamera = 1;
+  }
+  
+  public void cameraInit(){
+    new Thread(()-> {
+      Robot.m_camera.setCamera(Robot.m_camera.getCamera(1));
+    }).start();
+  }
+
+  public VideoSink getServer(){
+    return server;
+  }
+
+  public void setCamera(UsbCamera camera){
+    server.setSource(camera);
```

Comment:
Move the "whichCamera" setting to this function. It seems we always want to update "whichCamera" to the camera we set.

NOTE: You'll need to cleanup lines 45, 75, 77 to here.

Reply:
i think we handled that in the toggle camera function?

`public void toggleCameras() {
    if(whichCamera == 1){
      setCamera(m_camera2); 
      whichCamera = 2; // here we set the camera to 2? 
    }else{
      setCamera(m_camera1);
      whichCamera = 1; // and here to 1?
    }
  }`

Reply:
We can discuss when I see you next time. Yes it's handled in the toggleCameras() function but it's better to not have duplicate code.

Moving the "whichCamera" update into the "setCamera()" would ensure future "setCamera()" calls always appropriately update the "whichCamera" variable. Otherwise, you have to always update it separately to make sure you don't have a bug - so this screams refactoring to mitigate future bugs and to clean up the code.